ducktape/tests/loader.py

```diff
@@ -446,31 +452,52 @@ def _is_test_function(self, function):
 
     def _load_test_suite_files(self, test_suite_files):
         suites = list()
-        for test_suite_file_path in test_suite_files:
-            suites.extend(self._read_test_suite_from_file(test_suite_file_path))
+        
+        suites.extend(self._read_test_suite_from_file(test_suite_files))
 
         all_contexts = set()
         for suite in suites:
             all_contexts.update(self._load_test_suite(**suite))
         return all_contexts
 
-    def _read_test_suite_from_file(self, test_suite_file_path):
-        suites = list()
-        test_suite_file_path = os.path.abspath(test_suite_file_path)
-        if not os.path.exists(test_suite_file_path) or not os.path.isfile(test_suite_file_path):
-            raise LoaderException("Test suite does not exist or is a directory: " + test_suite_file_path)
-
-        with open(test_suite_file_path) as fp:
+    def _load_suite(self, suite_file_path):
+        with open(suite_file_path) as fp:
             try:
                 file_content = yaml.load(fp, Loader=yaml.FullLoader)
             except Exception as e:
-                raise LoaderException("Failed to load test suite from file: " + test_suite_file_path, e)
-
-            if not file_content:
-                raise LoaderException("Test suite file is empty: " + test_suite_file_path)
-            if not isinstance(file_content, dict):
-                raise LoaderException("Malformed test suite file: " + test_suite_file_path)
-
+                raise LoaderException("Failed to load test suite from file: " + suite_file_path, e)
+
+        if not file_content:
+            raise LoaderException("Test suite file is empty: " + suite_file_path)
+        if not isinstance(file_content, dict):
+            raise LoaderException("Malformed test suite file: " + suite_file_path)
+
+        for suite_name, suite_content in file_content.items():
+            if not suite_content:
+                raise LoaderException("Empty test suite " + suite_name + " in " + suite_file_path)
+        return file_content
+
+    def _read_test_suite_from_file(self, root_suite_file_paths):
+        suites = []
+        root_suite_file_paths = [os.path.abspath(file_path) for file_path in root_suite_file_paths]
+        files = {file: self._load_suite(file) for file in root_suite_file_paths}
+        stack = root_suite_file_paths
+
+        while len(stack) != 0:
+            curr = stack.pop()
+            loaded = files[curr]
+            if 'import' in loaded:
+                directory = os.path.dirname(curr)
+                # apply path of current file to the files inside
+                abs_file_iter = (os.path.abspath(os.path.join(directory, file))
```

Comment:
what if my test suite imports by absolute path?
```
import:
 - /Users/test/suite.yml
```
?

Reply:
`os.path.join` will use the second parameter only if it's an absolute path, so this should work. Testing is a bit more annoying since the yaml files can't be pre-baked...

Reply:
yeah not sure how I'm supposed to test absolute paths.

Reply:
There would be a couple of options I can think of:

1) Construct yamls dynamically instead of statically and use a tmp directory
2) Mock the file loading code to redirect as long as the expected absolute path matched
3) Decide it's not worth the testing effort and skip those tests. I'm not sure we even *should* support absolute paths. The only reason I can think of to support them is if you need to get a test suite from another project (e.g. CP tests use kafka tests), but I don't think we've planned or designed around that possibility.

Reply:
I have added a dynamic absolute path test.